packages/engine/src/utils.rs

```diff
@@ -73,11 +145,29 @@ impl Default for OutputFormat {
     }
 }
 
-pub fn init_logger(
-    std_err_output_format: OutputFormat,
+/// Guard for file logging, which should not be dropped until every log entry has been written.
+pub struct LogGuard {
```

Comment:
I do wonder if this would have been cleaner as a collection. Let's leave as is, as you've pointed out, it does remove a few empty Drop impls on our side